state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1814,6 +1927,44 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			return Ok(())
 		}
 
+		let channel_owner = deposit_channel_details.owner.clone();
+
+		if let Some(prewitness_state) =
+			TaintedTransactions::<T, I>::take(&channel_owner, &deposit_details)
+		{
+			if deposit_channel_details.boost_status == BoostStatus::NotBoosted ||
```

Comment:
I think this would benefit from a comment to explain the logic. Something like: "We refund if (a) we rejected a boost of this deposit earlier, or (b) if there has been no attempt to boost in the first place".

Alternatively, it may be simpler to understand the code if we recorded deposits that *have* boosted, then you could have a single condition here. I'm a little worried about relying on `boost_status` since it is about a channel rather than a deposit (we may have to consider cases where there are multiple desposits into the same channel). However, I realise that this isn't a trivial change, so may not be worth it.

Reply:
I agree that this is not very intuitive. All the information if a deposit was boosted or not is there, though. I think what could help to make it more understandable is first adding a comment that explains the code (like you suggested) and break the code down a bit with proper naming of the single conditions like that:
```
let is_not_boosted = deposit_channel_details.boost_status == BoostStatus::NotBoosted;
let was_tainted_before_boosting = prewitness_state == Prewitnessed::Yes;

if is_not_boosted || was_tainted_before_boosting {
    // ... (rest of the code)
}
```

Reply:
> was_tainted_before_boosting

This is still misleading though... If it is tainted before it is prewitnessed, then we *don't* boost it. IMO "boost rejected" (like I suggested above) would be more accurate. And sorry for being nitpicky about naming/comments 🙂, it is just that it is not easy to reason about both boosting and tainting at the same time, so I think in this particular case it is worth being more accurate about names to make reasoning about the code a little easier.